system/src/system_control_internal.cpp

```diff
@@ -389,6 +408,32 @@ void system_ctrl_set_result(ctrl_request* req, int result, ctrl_completion_handl
     particle::system::SystemControl::instance()->setResult(req, result, handler, data);
 }
 
+int system_ctrl_add_request_filter(uint16_t* reqItems, size_t cnt, void* reserved) {
+    // check if the element is already there. Check if the element is a valid element
+    for (unsigned i=0; i<cnt; i++) {
+        bool eleExists = false;
+        for (auto t : particle::system::SystemControl::instance()->vecCtrlReq_) {
```

Comment:
`vecCtrlReq_.contains()`. Let's also not access the vector directly from here and delegate to a class method instead.